code/modules/antagonists/morph/morph.dm

```diff
@@ -214,6 +214,11 @@
 					"<span class='notice'>You twist your body and assume the form of [target].</span>")
 	appearance = target.appearance
 	copy_overlays(target)
+	if(target.vis_contents)
+		for(var/I in 1 to length(target.vis_contents))
+			var/obj/p = target.vis_contents[I]
+			icon = p.icon
+			add_overlay(list(p.icon_state, p.overlays))
```

Comment:
```suggestion
	copy_overlays(target, TRUE)
	if(length(target.vis_contents))
		for(var/atom/A as() in target.vis_contents)
			add_overlay(A)
			copy_overlays(A)
```

Ideally, we should control all `vis_contents` via `SSvis_ovevrlays` and copy `vis_contents` with it, but this deep appearance copying is currently a better solution.

Reply:
Yeah that looks better trough i do not think the second copy_overlays is needed as add_overlay(A) already copies the icon and the overlays of an object.
At least from what i found out trough testing.

Reply:
Also using copy_overlays(A) will override the list as it looks like and thus deleting all the overlays already applied resulting the morph to become invisible again.